.github/workflows/ci_tests.yml

```
@@ -0,0 +1,86 @@
+name: CI Tests
+
+on:
+  push:
+  pull_request:
+  schedule:
+    # run every Monday at 5am UTC
+    - cron: '0 5 * * 1'
+
+env:
+  TOXARGS: '-v'
+
+jobs:
+  tests:
+    name: ${{ matrix.name }}
+    runs-on: ${{ matrix.os }}
+    if: "!(contains(github.event.head_commit.message, '[skip ci]') || contains(github.event.head_commit.message, '[ci skip]'))"
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+
+          - name: Code style checks
+            os: ubuntu-latest
+            python: 3.x
+            toxenv: codestyle
+
+          - name: docs build
+            os: ubuntu-latest
+            python: 3.8
+            toxenv: build_docs
+
+          - name: oldest dependencies
+            os: ubuntu-latest
+            python: 3.6
+            toxenv: py36-test-oldestdeps
+
+          - name: astropy dev with all dependencies with coverage
+            os: ubuntu-latest
+            python: 3.9
+            toxenv: py39-test-alldeps-devastropy-cov
```

I don't think you can use Python 3.9 with `alldeps` until all your dependencies have wheels for Python 3.9. See https://github.com/astropy/astroquery/pull/1925/files#r552092241 

```suggestion
            python: 3.8
            toxenv: py38-test-alldeps-devastropy-cov
```

---

I'm rather hacking the all dependencies around, but keep the widest possible testing on py39. Currently both aplpy and pyregion cause trouble.